platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/fan_drawer.py

```diff
@@ -2,30 +2,41 @@
     from sonic_platform.platform_thrift_client import thrift_try
     from sonic_platform_base.fan_drawer_base import FanDrawerBase
     from sonic_platform_base.fan_base import FanBase
+    from sonic_py_common import device_info
     from fan import Fan
 except ImportError as e:
     raise ImportError (str(e) + "- required module not found")
 
-_MAX_FAN = 10
+# Default Montara
+_MAX_FAN = 5
+_MAX_FANTRAY = 1
+
+_product_dict = {
+    "x86_64-accton_wedge100bf_32x-r0"   : "Montara",
+    "x86_64-accton_as9516_32d-r0"       : "Newport",
+    "x86_64-accton_as9516bf_32d-r0"     : "Newport",
+    "x86_64-accton_wedge100bf_65x-r0"   : "Mavericks"
+}
 
 def _fan_info_get_all():
     for fan_num in range(1, _MAX_FAN + 1):
         def get_data(client, fan_num=fan_num):
             return client.pltfm_mgr.pltfm_mgr_fan_info_get(fan_num)
-        fan_info = thrift_try(get_data)
+        fan_info = thrift_try(get_data)        
         if fan_info.fan_num == fan_num:
             yield fan_info
 
 # FanDrawer -> FanDrawerBase -> DeviceBase
 class FanDrawer(FanDrawerBase):
-    def __init__(self, fantray_index):
+    def __init__(self, fantray_index, modelstr="Montara"):
         # For now we return only present fans
-        self._fan_list = [Fan(i.fan_num) for i in _fan_info_get_all()]
+        self.modelstr = modelstr
         self.fantrayindex = fantray_index + 1
+        self._fan_list = [Fan(i.fan_num, self.fantrayindex, modelstr) for i in _fan_info_get_all()]
```

Comment:
it's overfill to call thrift API just to get FAN index
can't we just do `for i in range(1, _MAX_FAN + 1)` ?
note, better to pass num of fans as FanDrawer parameter instead of global var

Reply:
Looks like _fan_info_get_all() can be removed at all. Please check.